internal/pkg/cli/interfaces.go

```diff
@@ -498,3 +498,7 @@ type tasksStopper interface {
 type serviceLinkedRoleCreator interface {
 	CreateECSServiceLinkedRole() error
 }
+
+type dockerEngineValidator interface {
+	IsDockerEngineRunning() (string, error)
```

Comment:
nit: I'd expect a public method called IsDockerEngineRunning to return a boolean yes/no, not the error message. Can we change either the return value or the name? Something like `CallDockerEngine`

Reply:
Maybe `IsDockerEngineRunning() (bool, string, error)`? What do you think?

Reply:
What do you think of this:
```go
IsDockerEngineRunning(w io.Writer) (bool, error)
```
and we return specific error types for when the string was not empty, for example in a command:
```go
if err != nil {
   var errCmd *exec.ErrDockerCommandNotFound
   var errDaemon *exec.ErrDockerDaemonNotResponsive
   switch {
   case errors.As(err, &errCmd):
   case errors.As(err &errDaemon):
   default:
   }
}
```

Reply:
SGTM!